models/dataset.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Apr 14 20:36:19 2021
+
+@author: NLEGENDN
+"""
+import os
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, random_split
+import functools
+import h5py
+from PIL import Image
+CUDA = torch.cuda.is_available()
+
+if CUDA:
+    CRYO_TRAIN_VAL_DIR = os.getcwd()+"/Cryo/VAE_Cryo_V3/Data/"
+else:
+    CRYO_TRAIN_VAL_DIR = os.getcwd()+"\\Data\\"
+
+
+KWARGS = {'num_workers': 1, 'pin_memory': True} if CUDA else {}
+
+
+def open_dataset(path, new_size, D3):
+    """
+    Parameters
+    ----------
+    path : string path.
+    new_size : int.
+    D3 : boolean if 2d or 3d
+
+    Returns
+    -------
+    tensor of images in black and white.
+
+    """
+    if not os.path.exists(path):
+        raise OSError
+    if path[-3:] == '.h5':
+        data_dict = h5py.File(path, 'r')
+        all_datasets = data_dict['particles'][:]
+    else:
+        all_datasets = np.load(path)
+    dataset = np.asarray(all_datasets)
+    img_shape = dataset.shape
+    N = img_shape[0]
+    Dataset = []
+    if D3:
+        dataset = torch.Tensor(dataset)
+        dataset = normalization_linear(dataset)
+    else:
+        if len(img_shape) == 3:
+            for i in range(N):
+                Dataset.append(np.asarray(Image.fromarray(
+                    dataset[i]).resize([new_size, new_size])))
+        elif len(img_shape) == 4:
+            for i in range(N):
+                Dataset.append(np.asarray(Image.fromarray(
+                    dataset[i][0]).resize([new_size, new_size])))
+        dataset = torch.Tensor(Dataset)
+        dataset = normalization_linear(dataset)
+        if len(img_shape) != 4:
+            dataset = dataset.reshape(
+                (img_shape[0], 1, img_shape[1], img_shape[1]))
+    return dataset
+
+
+def normalization_linear(dataset):
+    """
+    Parameters
+    ----------
+    dataset : tensor of images.
+
+    Returns
+    -------
+    dataset : normalized tensor of images.
+
+    """
+    for i, data in enumerate(dataset):
+        min_data = torch.min(data)
+        max_data = torch.max(data)
+        if max_data != min_data:
+            raise ZeroDivisionError
+        dataset[i] = (data - min_data) / (max_data - min_data)
+    return dataset
+
+
+def organize_dataset(dataset, batch_size, frac_val):
+    """
+    Parameters
+    ----------
+    dataset : tensor of images.
+    batch_size : int batch_size.
+    frac_val : ratio between validation and training datasets.
+
+    Returns
+    -------
+    trainset : tensor of training images.
+    testset : tensor of test images.
+    trainloader : tensor ready to be used by the NN for training images.
+    testloader : tensor ready to be used by the NN for test images.
+
+    """
+    N = len(dataset)
+    n = int(N*frac_val)
```

Comment:
n_val

Reply:
(try to use variable names that will help the collaborators to understand your code)